Generate new expected results file when there is failures in diff time benchmarks

laithsakka opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #137551
The test also add singpost log for the benchmarks that pass.
to test run I ran python check_results.py test_check_result/expected_test.csv test_check_result/result_test.csv out.csv
results
```
WIN: benchmark ('a', 'instruction count') failed, actual result 90 is -18.18% lower than expected 110 ±1.00% please update the expected results.

REGRESSION: benchmark ('b', 'memory') failed, actual result 200 is 100.00% higher than expected 100 ±+10.00% if this is an expected regression, please update the expected results.

PASS: benchmark ('c', 'something') pass, actual result 107 +7.00% is within expected 100 ±10.00%

MISSING REGRESSION TEST: benchmark ('d', 'missing-test') does not have a regression test enabled for it.

You can use the new reference expected result stored at path: out.csv.

a,instruction count,90,0.01
b,memory,200,0.1
c,something,100,0.1
```


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

laithsakka committed:
generate new expected results file when there is faluires in diff time benchmarks

[ghstack-poisoned]

laithsakka committed:
Update on "generate new expected results file when there is faluires in diff time benchmarks"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka committed:
Update on "generate new expected results file when there is faluires in diff time benchmarks"

to test run 
python check_results.py test_check_result/expected_test.csv test_check_result/result_test.csv out.csv
results
```
WIN: benchmark ('a', 'instruction count') failed, actual result 90 is -18.18% lower than expected 110 ±1.00% please update the expected results.

REGRESSION: benchmark ('b', 'memory') failed, actual result 200 is 100.00% higher than expected 100 ±10.00% if this is an expected regression, please update the expected results.

PASS: benchmark ('c', 'something') pass, actual result 107 -7.00% is within expected 100 ±10.00%

MISSING REGRESSION TEST: benchmark ('d', 'missing-test') does not have a regression test enabled for it.

You can use the new reference expected result stored at path: out.csv.

a,instruction count,110,0.01
b,memory,100,0.1
c,something,100,0.1
```


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka committed:
Update on "generate new expected results file when there is faluires in diff time benchmarks"

to test run 
python check_results.py test_check_result/expected_test.csv test_check_result/result_test.csv out.csv
results
```
WIN: benchmark ('a', 'instruction count') failed, actual result 90 is -18.18% lower than expected 110 ±1.00% please update the expected results.

REGRESSION: benchmark ('b', 'memory') failed, actual result 200 is 100.00% higher than expected 100 ±10.00% if this is an expected regression, please update the expected results.

PASS: benchmark ('c', 'something') pass, actual result 107 -7.00% is within expected 100 ±10.00%

MISSING REGRESSION TEST: benchmark ('d', 'missing-test') does not have a regression test enabled for it.

You can use the new reference expected result stored at path: out.csv.

a,instruction count,110,0.01
b,memory,100,0.1
c,something,100,0.1
```


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka committed:
Update on "generate new expected results file when there is faluires in diff time benchmarks"

to test run 
python check_results.py test_check_result/expected_test.csv test_check_result/result_test.csv out.csv
results
```
WIN: benchmark ('a', 'instruction count') failed, actual result 90 is -18.18% lower than expected 110 ±1.00% please update the expected results.

REGRESSION: benchmark ('b', 'memory') failed, actual result 200 is 100.00% higher than expected 100 ±10.00% if this is an expected regression, please update the expected results.

PASS: benchmark ('c', 'something') pass, actual result 107 -7.00% is within expected 100 ±10.00%

MISSING REGRESSION TEST: benchmark ('d', 'missing-test') does not have a regression test enabled for it.

You can use the new reference expected result stored at path: out.csv.

a,instruction count,110,0.01
b,memory,100,0.1
c,something,100,0.1
```


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

aorenste commented on benchmarks/dynamo/pr_time_benchmarks/check_results.py:
```diff
@@ -92,34 +99,52 @@ def log(event_name):
+            ratio = float(entry.expected_value - result) * 100 / entry.expected_value
+
+            print(
+                f"PASS: benchmark {key} pass, actual result {result} {'+' if ratio>0 else ''}{ratio:.2f}% is within "
```
You can force the sign by adding `+` to the format:
```suggestion
                f"PASS: benchmark {key} pass, actual result {result} {ratio:+.2f}% is within "
```

aorenste commented on benchmarks/dynamo/pr_time_benchmarks/check_results.py:
```diff
@@ -92,34 +99,52 @@ def log(event_name):
+
+            print(
+                f"PASS: benchmark {key} pass, actual result {result} {'+' if ratio>0 else ''}{ratio:.2f}% is within "
+                f"expected {entry.expected_value} ±{entry.noise_margin*100:.2f}%\n"
```
Instead of adding non-ascii to the code use the unicode escape:
```suggestion
                f"expected {entry.expected_value} \u00b1{entry.noise_margin*100:.2f}%\n"
```
or maybe better:
```suggestion
            plus_minus_sign = "\u00b1"
             ...
                f"expected {entry.expected_value} {plus_minus_sign}{entry.noise_margin*100:.2f}%\n"
```

laithsakka replied:
why?

aorenste reviewed (approved):
approving with comments

laithsakka commented on benchmarks/dynamo/pr_time_benchmarks/check_results.py:
```diff
@@ -92,34 +99,52 @@ def log(event_name):
                 ),
             )
 
+        ratio = float(result - entry.expected_value) * 100 / entry.expected_value
```
ye it does but its ok , as shown the log message, in summary .

laithsakka replied:
1.  -18.18% lower vs 18.18% lower is ok 
2. i do not need the recompute my bad.

laithsakka commented on benchmarks/dynamo/pr_time_benchmarks/check_results.py:
```diff
@@ -35,6 +36,11 @@ def main():
     result_file_path = sys.argv[2]
 
+    # A path where a new expected results file will be written that can be used to replace expected_results.csv
+    # in case of faluire. In case of no faluire the content of this file will match expected_file_path, values
```
i installed spell checker multiple times, not sure where does it go lol I wonder . lol

laithsakka commented on benchmarks/dynamo/pr_time_benchmarks/check_results.py:
```diff
@@ -68,6 +74,7 @@ def main():
             result_data[key] = entry
 
     fail = False
+    new_expected = copy.deepcopy(expected_data)
```
ah should be used line 167 lol

laithsakka committed:
Update on "Generate new expected results file when there is failures in diff time benchmarks"

The test also add singpost log for the benchmarks that pass.
to test run I ran python check_results.py test_check_result/expected_test.csv test_check_result/result_test.csv out.csv
results
```
WIN: benchmark ('a', 'instruction count') failed, actual result 90 is -18.18% lower than expected 110 ±1.00% please update the expected results.

REGRESSION: benchmark ('b', 'memory') failed, actual result 200 is 100.00% higher than expected 100 ±10.00% if this is an expected regression, please update the expected results.

PASS: benchmark ('c', 'something') pass, actual result 107 -7.00% is within expected 100 ±10.00%

MISSING REGRESSION TEST: benchmark ('d', 'missing-test') does not have a regression test enabled for it.

You can use the new reference expected result stored at path: out.csv.

a,instruction count,110,0.01
b,memory,100,0.1
c,something,100,0.1
```


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka commented:
@pytorchbot merge

eellison commented on benchmarks/dynamo/pr_time_benchmarks/check_results.py:
```diff
@@ -92,34 +99,50 @@ def log(event_name):
-                f"is {ratio:.2f}% higher than expected {entry.expected_value} ±{entry.noise_margin*100:.2f}% "
-                f"if this is an expected regression, please update the expected results."
+                f"is {ratio:.2f}% higher than expected {entry.expected_value} ±{entry.noise_margin*100:+.2f}% "
+                f"if this is an expected regression, please update the expected results.\n"
```
I know it's listed below, but would be nice if there were instructions on how to update in the msg, like with other pytorch failures that tell you to run `EXPECTTEST=1 {test}`. The ci csv is not easy to understand where to get or regenerate
